### PR TITLE
pagination_component

### PR DESCRIPTION
**Pull request by yash6102000**

### CODE REVIEW

File: app/components/common_components/pagination/Pagination.tsx [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+}: {
+  onPageChange: (e: number) => void
+  totalLength: number
+  currentPage: number
+  pageSize: number
+  setPageSize: (e: number) => void
+  firstPageIndex: number
+  testDataLength: number
+}) => {
+  // const totalCount = Math.ceil(totalLength / pageSize)
```
Comment (richa-copods) [inline app/components/common_components/pagination/Pagination.tsx:23]: Remove unwanted comments

File: app/components/common_components/pagination/Pagination.tsx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+  const siblingCount = 1
+  const paginationRange: any[] | undefined = usePagination({
```
Comment (richa-copods) [inline app/components/common_components/pagination/Pagination.tsx:25]: Give proper type. Avoid using any

File: app/components/common_components/pagination/Pagination.tsx [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+              if (e.key === 'Enter') onPageChange(pageNumber)
+            }}
+          >
+            {pageNumber}
+          </span>
+        )
+      }
+    })
+  }
+  const itemsList = [
```
Comment (richa-copods) [inline app/components/common_components/pagination/Pagination.tsx:61]: It should be passed from parent and should be dynamic

File: app/components/common_components/pagination/Pagination.tsx [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    { name: '5 Items', pageSize: 5 },
+    { name: '10 Items', pageSize: 10 },
+    { name: '15 Items', pageSize: 15 },
+    { name: '20 Items', pageSize: 20 },
+  ]
+  const [selected, setSelected] = useState(itemsList[0])
+  const dropDown = () => {
+    return (
+      <div className="">
```
Comment (richa-copods) [inline app/components/common_components/pagination/Pagination.tsx:70]: Remove classname

File: app/components/common_components/pagination/Pagination.tsx [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+                ))}
+              </Listbox.Options>
+            </Transition>
+          </div>
+        </Listbox>
+      </div>
+    )
+  }
+  return (
+    <>
```
Comment (richa-copods) [inline app/components/common_components/pagination/Pagination.tsx:113]: Fragment isn't needed here

File: app/components/common_components/pagination/usePagination.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
+import { useMemo } from 'react'
+export const DOTS = '...'
+
```
Comment (richa-copods) [inline app/components/common_components/pagination/usePagination.js:3]: Change file type to tsx and define types

File: app/components/common_components/pagination/usePagination.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+export const usePagination = ({
+  totalLength,
+  pageSize,
+  siblingCount = 1,
+  currentPage,
+}) => {
+  const paginationRange = useMemo(() => {
+    const totalPageCount = Math.ceil(totalLength / pageSize)
+    const totalPageNumbers = siblingCount + 5
```
Comment (richa-copods) [inline app/components/common_components/pagination/usePagination.js:12]: Why hardcoding totalPageNumbers?

File: app/components/common_components/pagination/usePagination.js [hunk 8]
```diff
@@ -0,0 +1,49 @@
+import { useMemo } from 'react'
```
Comment (richa-copods) [inline app/components/common_components/pagination/usePagination.js:1]: What is the purpose of this file?

File: app/components/SideNavFooter.tsx [hunk 9]
```diff
@@ -1,7 +1,7 @@
 import { Icon } from '@iconify/react'
 import { useUser } from '~/utils'
 import { Form } from '@remix-run/react'
-import Button from './form/Button'
+import Button from './common_components/Button'
```
Comment (richa-copods) [inline app/components/SideNavFooter.tsx:4]: Component folders' names should be kebab-case. Follow ReadME.md file

**Review by richa-copods — CHANGES_REQUESTED** [review]

File: app/components/common-components/pagination/Pagination.tsx [hunk 10]
```diff
@@ -24,7 +24,7 @@ const Pagination = ({
 }) => {
   const totalCount = Math.ceil(totalLength / pageSize)
   const siblingCount = 1
-  const paginationRange: any[] | undefined = usePagination({
+  const paginationRange: Array<any> | undefined = usePagination({
```
Comment (akshay-copods) [inline app/components/common-components/pagination/Pagination.tsx:27]: Please give proper type.

**Review by akshay-copods — CHANGES_REQUESTED** [review]

**Review by akshay-copods — APPROVED** [review]

**Review by anuragP04 — APPROVED** [review]

**Pull request merged**